Add missing server variable to OpenAPI (#11)

praphull opened the pull request:
Add missing field 'variables' in Server class from OpenAPI spec. Doesn't handle the restriction of "default" being one of the values in "enum", if "enum" is defined.
Not sure where you'd want the restriction to be applied as per this project's conventions - e.g. maybe throw an exception in constructor of ServerVariable upon violation?

P.S. This is my first ever PR on a proper project, so do let know how if the changes aren't sufficient or something's off

adamw commented:
Looks good! :)

If there are some restrictions which we cannot capture in types, it's better to throw an exception than do nothing. A usual `require` in the constructor should do, see for example: https://github.com/softwaremill/tapir/blob/master/core/src/main/scala/sttp/tapir/EndpointIO.scala#L116

Thanks!

praphull commented:
> Looks good! :)
> 
> If there are some restrictions which we cannot capture in types, it's better to throw an exception than do nothing. A usual `require` in the constructor should do, see for example: https://github.com/softwaremill/tapir/blob/master/core/src/main/scala/sttp/tapir/EndpointIO.scala#L116
> 
> Thanks!

Thanks, I've updated it as mentioned.

Btw, the CI build failed for no apparent reason when I pushed my commit earlier, so I rebased over master and force pushed - which I now understand wasn't required.

adamw commented:
Released in 0.12.24, thanks!
